src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java

```diff
@@ -145,13 +150,22 @@ public boolean start() throws Exception {
                 @Override
                 public void run() {
                     try {
-                        CIMessageSubscriberBuilder subscriber = new CIMessageSubscriberBuilder(
-                                step.getProviderName(),
-                                step.getOverrides(),
-                                step.getSelector(),
-                                step.getChecks(),
-                                step.getTimeout()
-                                );
+                        ProviderData pd = null;
+                        JMSMessagingProvider p = GlobalCIConfiguration.get().getProvider(step.getProviderName());
```

Comment:
if you have the provider, it should be able to give you the providerData. we should not have to check for `instanceof` here.